packages/@n8n/ai-workflow-builder.ee/src/prompts/chains/parameter-updater/guides/system-message.ts

```diff
@@ -1,4 +1,42 @@
-export const SYSTEM_MESSAGE_GUIDE = `
+import type { NodeTypeGuide } from '../types';
+
+/**
+ * Checks if a node has system message parameters.
+ * Applies to AI Agent, LLM Chain, Anthropic, OpenAI, etc.
+ */
+function hasSystemMessageParameters(nodeDefinition: {
+	properties?: Array<{ name: string; type: string; options?: unknown[] }>;
+}): boolean {
+	if (!nodeDefinition.properties) return false;
+
+	return nodeDefinition.properties.some((prop) => {
+		// Pattern 1 & 2: options.systemMessage (AI Agent) or options.system (Anthropic)
+		if (prop.name === 'options' && prop.type === 'collection') {
```

Comment:
Nit: double if statement is a little less readable here - I'd love the pattern 3 check first to demonstrate always true scenarios up front, then pattern 1 & 2 collapse into a single if, like:
```ts
	// Pattern 1: messages parameter with role support
	if (
		prop.name === 'messages' &&
		(prop.type === 'fixedCollection' || prop.type === 'collection')
	) {
		return true;
	}

	// Pattern 2 & 3: options.systemMessage / options.system
	if (prop.name === 'options' && prop.type === 'collection' && Array.isArray(prop.options)) {
		return prop.options.some(
			(opt) =>
				typeof opt === 'object' &&
				opt !== null &&
				'name' in opt &&
				(opt.name === 'systemMessage' || opt.name === 'system'),
		);
	}

    return false;
```